{
  "comments": [
    {
      "unresolved": true,
      "key": {
        "uuid": "da1a8d1a_2a2cf56f",
        "filename": "plays/upgrade-csv.yml",
        "patchSetId": 1
      },
      "lineNbr": 9,
      "author": {
        "id": 667
      },
      "writtenOn": "2023-02-03T17:47:46Z",
      "side": 1,
      "message": "There\u0027s a typo in the namespace",
      "revId": "2eafec1f571386137fbf924e725980e769f3273a",
      "serverId": "6c1dc8ef-8b94-40e4-bd83-c2359d45ecc0"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "2cf21184_53963d37",
        "filename": "plays/upgrade-csv.yml",
        "patchSetId": 1
      },
      "lineNbr": 9,
      "author": {
        "id": 789
      },
      "writtenOn": "2023-02-06T09:32:03Z",
      "side": 1,
      "message": "nice catch. Thanks a lot.",
      "parentUuid": "da1a8d1a_2a2cf56f",
      "revId": "2eafec1f571386137fbf924e725980e769f3273a",
      "serverId": "6c1dc8ef-8b94-40e4-bd83-c2359d45ecc0"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "ce77da38_1ee6171e",
        "filename": "plays/upgrade-csv.yml",
        "patchSetId": 1
      },
      "lineNbr": 9,
      "author": {
        "id": 667
      },
      "writtenOn": "2023-02-15T19:34:45Z",
      "side": 1,
      "message": "Done",
      "parentUuid": "2cf21184_53963d37",
      "revId": "2eafec1f571386137fbf924e725980e769f3273a",
      "serverId": "6c1dc8ef-8b94-40e4-bd83-c2359d45ecc0"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "2f704bc1_b3a1a238",
        "filename": "plays/upgrade-csv.yml",
        "patchSetId": 1
      },
      "lineNbr": 10,
      "author": {
        "id": 667
      },
      "writtenOn": "2023-02-03T17:47:46Z",
      "side": 1,
      "message": "Could you try and see if by setting\n\n    field_selectors:\n      - status.catalogSource \u003d \u0027{{ sub.catalogSource }}\u0027\n\nwould filter and give you the packagemanifest you want?",
      "revId": "2eafec1f571386137fbf924e725980e769f3273a",
      "serverId": "6c1dc8ef-8b94-40e4-bd83-c2359d45ecc0"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "6b52fdb4_d3813e69",
        "filename": "plays/upgrade-csv.yml",
        "patchSetId": 1
      },
      "lineNbr": 10,
      "author": {
        "id": 789
      },
      "writtenOn": "2023-02-06T09:32:03Z",
      "side": 1,
      "message": "I tried this when researching the solution by running the oc command and it always complained that status.catalogSource was not a known field for searching, and only name and namespace were supported:\n\n```\n$ oc -n vran-acceleration-operators get packagemanifest --field-selector\u003dstatus.catalogSource\u003dintel-operators\nError from server (BadRequest): Unable to find \"packages.operators.coreos.com/v1, Resource\u003dpackagemanifests\" that match label selector \"\", field selector \"status.catalogSource\u003dintel-operators\": \"status.catalogSource\" is not a known field selector: only \"metadata.name\", \"metadata.namespace\"\n```",
      "parentUuid": "2f704bc1_b3a1a238",
      "revId": "2eafec1f571386137fbf924e725980e769f3273a",
      "serverId": "6c1dc8ef-8b94-40e4-bd83-c2359d45ecc0"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "dafa75f1_2cc70a27",
        "filename": "plays/upgrade-csv.yml",
        "patchSetId": 1
      },
      "lineNbr": 10,
      "author": {
        "id": 667
      },
      "writtenOn": "2023-02-06T14:22:08Z",
      "side": 1,
      "message": "Did you try with the ansible module? The doc suggest that it is possible: https://github.com/ansible-collections/community.kubernetes/blob/stable-1.2/plugins/modules/k8s_info.py#L93-L97",
      "parentUuid": "6b52fdb4_d3813e69",
      "revId": "2eafec1f571386137fbf924e725980e769f3273a",
      "serverId": "6c1dc8ef-8b94-40e4-bd83-c2359d45ecc0"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "a1c03e6a_46cae8bd",
        "filename": "plays/upgrade-csv.yml",
        "patchSetId": 1
      },
      "lineNbr": 10,
      "author": {
        "id": 667
      },
      "writtenOn": "2023-02-13T20:04:41Z",
      "side": 1,
      "message": "I forgot if you tested this or not?",
      "parentUuid": "dafa75f1_2cc70a27",
      "revId": "2eafec1f571386137fbf924e725980e769f3273a",
      "serverId": "6c1dc8ef-8b94-40e4-bd83-c2359d45ecc0"
    },
    {
      "unresolved": true,
      "key": {
        "uuid": "958e81e7_3aa6dd34",
        "filename": "plays/upgrade-csv.yml",
        "patchSetId": 1
      },
      "lineNbr": 10,
      "author": {
        "id": 789
      },
      "writtenOn": "2023-02-15T09:12:07Z",
      "side": 1,
      "message": "Hi Tony,\n\nJust tested your suggestion. \n\nI used the following code:\n\n```\n- hosts: localhost\n  vars:\n    sub:\n      operatorName: sriov-fec\n      catalogSource: intel-operators\n      namespace: vran-acceleration-operators\n  tasks:\n    - k8s_info:\n        kubeconfig: /var/lib/dci-openshift-agent/clusterconfigs-cluster1/kubeconfig\n        api: packages.operators.coreos.com/v1\n        kind: PackageManifest\n        namespace: \"{{ sub.namespace }}\"\n        field_selectors:\n          - status.catalogSource \u003d \u0027{{ sub.catalogSource }}\u0027\n      register: packagemanifest_all\n      retries: 10\n      delay: 5\n      until:\n        - packagemanifest_all.resources is defined\n        - packagemanifest_all.resources | length \u003e 0\n\n    - debug:\n        msg: \"resources found: {{ packagemanifest_all.resources | lenght }}\"\n```\n\nWhich resulted in the following error output. I\u0027m omitting most of the content but the idea is that we get the same unknown field error message:\n\n```\n$ ansible-playbook /tmp/select.yml                                                                                                           \n[WARNING]: provided hosts list is empty, only localhost is available. Note that the                                                                                          \nimplicit localhost does not match \u0027all\u0027                                                                                                                                      \n                                                                                                                                                                             \nPLAY [localhost] *********************************************************************                                                                                       \n                                                                                                                                                                             \nTASK [Gathering Facts] ***************************************************************                                                                                       \nok: [localhost]                                                                                                                                                              \n                                                                                                                                                                             \nTASK [k8s_info] **********************************************************************                                                                                       \nFAILED - RETRYING: k8s_info (10 retries left).                                                                                                                               \n[...]                                                                                                                                \nFAILED - RETRYING: k8s_info (1 retries left).                                                                                                                                \nAn exception occurred during task execution. To see the full traceback, use -vvv. The error was:     raise ApiException(http_resp\u003dr)                                         \nfatal: [localhost]: FAILED! \u003d\u003e {\"attempts\": 10, \"changed\": false, \"module_stderr\": \"Traceback \n[...]\nReason: Bad Request\\nHTTP response headers: HTTPHeaderDict({\u0027Audit-Id\u0027: \u0027c2b90ad5-e754-43cc-9$15-edacf442190a, c2b90ad5-e754-43cc-9415-edacf442190a\u0027, \u0027Cache-Control\u0027: \u0027no-cache, private, no-cache, private\u0027, \u0027Content-Length\u0027: \u0027218\u0027, \u0027Content-Type\u0027: \u0027application/json\u0027$ \u0027Date\u0027: \u0027Tue, 07 Feb 2023 09:51:19 GMT\u0027, \u0027X-Kubernetes-Pf-Flowschema-Uid\u0027: \u00271243f5a2-40c6-40f9-9aa7-6a844c389f6b\u0027, \u0027X-Kubernetes-Pf-Prioritylevel-Uid\u0027: \u00273b655d97-2d6b-435c$9597-184865aeea6a\u0027})\\nHTTP response body: b\u0027{\\\"kind\\\":\\\"Status\\\",\\\"apiVersion\\\":\\\"v1\\\",\\\"metadata\\\":{},\\\"status\\\":\\\"Failure\\\",\\\"message\\\":\\\"\\\\\\\\\\\"status.catalogSource \\\\\\\\\\$ is not a known field selector: only \\\\\\\\\\\"metadata.name\\\\\\\\\\\", \\\\\\\\\\\"metadata.namespace\\\\\\\\\\\"\\\",\\\"reason\\\":\\\"BadRequest\\\",\\\"code\\\":400}\n[...]\n```\nAll the same, I\u0027m going to include the following field selector which will return the two sriov-fec package manifests in opposition to setting \"name: sriov-fec\" which only returns one, thus the original problem.\n\n```\nfield_selectors:\n  - metadata.name \u003d \"{{ sub.operatorName }}\"\n```",
      "parentUuid": "dafa75f1_2cc70a27",
      "revId": "2eafec1f571386137fbf924e725980e769f3273a",
      "serverId": "6c1dc8ef-8b94-40e4-bd83-c2359d45ecc0"
    },
    {
      "unresolved": false,
      "key": {
        "uuid": "c7bed276_f2aef9a2",
        "filename": "plays/upgrade-csv.yml",
        "patchSetId": 1
      },
      "lineNbr": 10,
      "author": {
        "id": 667
      },
      "writtenOn": "2023-02-15T19:34:45Z",
      "side": 1,
      "message": "Done",
      "parentUuid": "958e81e7_3aa6dd34",
      "revId": "2eafec1f571386137fbf924e725980e769f3273a",
      "serverId": "6c1dc8ef-8b94-40e4-bd83-c2359d45ecc0"
    }
  ]
}